azurerm/resource_arm_iot_dps_cerificate.go

```diff
@@ -0,0 +1,149 @@
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_iot_dps_certificate", *existing.ID)
+		}
+	}
+
+	certificate := iothub.CertificateBodyDescription{
+		Certificate: utils.String(d.Get("certificate_content").(string)),
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, iotDPSName, name, certificate, ""); err != nil {
+		return fmt.Errorf("Error creating/updating IoT Device Provisioning Service Certificate %q (Device Provisioning Service %q / Resource Group %q): %+v", name, iotDPSName, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, name, resourceGroup, iotDPSName, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving IoT Device Provisioning Service Certificate %q (Device Provisioning Service %q / Resource Group %q): %+v", name, iotDPSName, resourceGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read IoT Device Provisioning Service Certificate %q (Device Provisioning Service %q / Resource Group %q): %+v", name, iotDPSName, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmIotDPSCertificateRead(d, meta)
+}
+
+func resourceArmIotDPSCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).iothub.DPSCertificateClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	iotDPSName := id.Path["provisioningServices"]
+	name := id.Path["certificates"]
+
+	resp, err := client.Get(ctx, name, resourceGroup, iotDPSName, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving IoT Device Provisioning Service Certificate %q (Device Provisioning Service %q / Resource Group %q): %+v", name, iotDPSName, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("iot_dps_name", iotDPSName)
+
+	return nil
+}
+
+func resourceArmIotDPSCertificateDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).iothub.DPSCertificateClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	iotDPSName := id.Path["provisioningServices"]
+	name := id.Path["certificates"]
+
+	resp, err := client.Get(ctx, name, resourceGroup, iotDPSName, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return nil
+		}
+		return fmt.Errorf("Error retrieving IoT Device Provisioning Service Certificate %q (Device Provisioning Service %q / Resource Group %q): %+v", name, iotDPSName, resourceGroup, err)
+	}
+
+	// TODO address this delete call if https://github.com/Azure/azure-rest-api-specs/pull/6311 get's merged
+	if _, err := client.Delete(ctx, resourceGroup, *resp.Etag, iotDPSName, name, "", nil, nil, iothub.ServerAuthentication, nil, nil, nil, ""); err != nil {
```

Comment:
since there's a potential panic here if `resp.Etag` is nil - could we nil-check this?